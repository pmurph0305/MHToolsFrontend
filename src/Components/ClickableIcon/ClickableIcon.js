import React from 'react'


/**
 * @param  {string} iconName ionicon icon name
 * @param  {string} iconSize ionicon icon size (small or large)
 * @param  {function} onClick onClick event handler
 */
const ClickableIcon = ({ iconName, iconSize, onClick }) => {
    return (
        <ion-icon
            name={iconName}
            onClick={onClick}
            size={iconSize ? iconSize : "default"}            
        />
    )
}

export default ClickableIcon;