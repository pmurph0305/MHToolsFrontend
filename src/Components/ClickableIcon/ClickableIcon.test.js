import React from "react";
import { shallow } from "enzyme";
import ClickableIcon from "./ClickableIcon";

describe("ClickableIcon tests", () => {
  let mockClick = jest.fn(val => val);
  let mockProps = {
    iconName: "share-alt",
    onClick: mockClick,
    iconSize: "small"
  };

  const wrapper = shallow(<ClickableIcon {...mockProps} />);

  it("Exists", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("Responds to clicks", () => {
    wrapper.find("ion-icon").simulate("click");
    expect(mockClick).toHaveBeenCalledTimes(1);
    wrapper.find("ion-icon").simulate("click");
    expect(mockClick).toHaveBeenCalledTimes(2);
  });

  it("Has correct properties", () => {
    expect(wrapper.find('[name="share-alt"]').length).toBe(1);
    expect(wrapper.find('[size="small"]').length).toBe(1);
  });

  it("Sets size to default if no size", () => {
    let mockProps2 = {
      iconName: "share-alt",
      onClick: mockClick
    };
    const defaultwrapper = shallow(<ClickableIcon {...mockProps2} />);
    expect(defaultwrapper.find('[size="default"]').length).toBe(1);
  });

  it("Matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
