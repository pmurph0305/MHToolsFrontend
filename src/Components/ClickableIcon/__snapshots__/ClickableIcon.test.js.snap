// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableIcon tests Matches snapshot 1`] = `ShallowWrapper {}`;
